File: binary_search.py
def binary_search(array, key):
    right = len(array) - 1
    left = 0

    while left <= right:
        mid = (left + right)//2
        
        if array[mid] == key:
            return mid
        elif array[mid] < key:
            left = mid + 1
        elif array[mid] > key:
            right = mid - 1
    return -1

arraylist = [2, 4, 7, 8, 12, 15, 24]
print(arraylist)
input = int(input("Enter the value you want to search for: "))
answer = binary_search(arraylist, input)
if answer != -1:
    print(f"{input} was found at index {answer}")
else:
    print("Not found")  

File: Binary_search.java
static int binary_search(int array[], int key){
    int right = array.length - 1;
    int left = 0;
    while(left <= right){
        int mid = left + (right - left) / 2;
        if(array[mid] == key)
            return mid;
        else if(array[mid] < key)
            left = mid + 1;
        else
            right = mid - 1;
    }
public class Binary_search {
    public static void main(String[] args){
        int[] my_array = {10, 20, 30, 40, 50};
        int key = 30;
        int result = binary_search(my_array, key);
        if(result != -1)
            System.out.println("Element found at index: " + result);
        else
            System.out.println("Element not found.");
    }
}

